analysis/runtime/src/runtime/scoped_runtime.rs

```diff
@@ -147,12 +168,14 @@ impl ExistingRuntime for BackgroundThreadRuntime {
     fn send_event(&self, event: Event) {
         match self.finalized.get() {
             None => {
-                // `.unwrap()` as we're in no place to handle an error here,
-                // unless we should silently drop the [`Event`] instead.
-                self.tx.send(event).unwrap();
+                // # Signal-safety: `push_event` is safe if `can_sleep == false`
+                self.push_event(event, false);
             }
             Some(()) => {
-                // Silently drop the [`Event`] as the [`BackgroundThreadRuntime`] has already been [`BackgroundThreadRuntime::finalize`]d.
+                // Silently drop the [`Event`] as the [`BackgroundThreadRuntime`]
+                // has already been [`BackgroundThreadRuntime::finalize`]d.
+                //
+                // # Signal-safety: Should be fine even after the end of `main`.
```

Comment:
Should we note that this is only true for `once_cell::OnceCell`'s (default) `feature = "std"` implementation?  `std::sync::OnceLock`, the supposed replacement, uses a `futex_wait`.

Also, while pre `main` there shouldn't be any signal handlers installed yet, post `main`, any signal handlers installed probably weren't uninstalled, so a post `main` signal seems more likely than a pre `main` one.

Reply:
Oh, nevermind, I see you've removed the code in `AfterMain`.